### ISSUE TITLE
Fix `invalid-attr` evaluates `noUse` property on an element schema

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#1734]
- Markuplint version: `4.x.x`
- Parser lang: HTML or JSX and more...
- Node.js version: `20.x.x`
- OS: all

## Describe the bug

https://github.com/markuplint/markuplint/blob/e6e78681f89d72aff061d75d170d43b48dda9984/packages/%40markuplint/html-spec/src/spec.dialog.json#L18-L21

## Code Example or Playground URL

```html
<dialog tabindex="0">
...
</dialog>
```

https://playground.markuplint.dev/#N4Igxg9gdgZglgcxALhMAOlABF9ICmAHgC75QAmAznslgNp4C2AhgE4DWArgA4A2cUYslb5IjRmXL5yeALqYAvphAAacBCkAxOL3wAVAJ7d8KEADoAFsUa9V6qaYA85OM14QEWNq4C0vZgBG+LwAvHgweFjEgQJShGEgAAx4AHyYmI4A9C5uHikgCkA

## The current behavior

No warning

## The expected behavior

Disallow the `tabindex` attribute on the `dialog` element

**Comment by YusukeHirao** [comment]
## Investigation: Already Fixed

This issue has already been resolved by the following commits:

- `50946c6ad` — Added a test for `noUse` flag
- `224996e77` — Implemented `noUse` property evaluation in `attr-check.ts` and `invalid-attr/index.ts`

### Current behavior

Confirmed that `<dialog tabindex="-1"></dialog>` correctly reports **"The "tabindex" attribute is disallowed"** via the `invalid-attr` rule.

### How it works

1. `html-spec/src/spec.dialog.json` defines `tabindex: { noUse: true }`
2. `ml-spec/src/utils/get-attr-specs-spec.ts` merges global attributes with element-specific attributes, preserving `noUse: true`
3. `rules/src/attr-check.ts` checks `spec.noUse` and returns a `disallowed-attr` violation
4. The test case (`noUse flag` in `invalid-attr/index.spec.ts`) passes

Closing as already fixed.

**Comment by YusukeHirao** [comment]
## Reopening: Bug is still present

My earlier analysis was incorrect. While the `noUse` check in `attrCheck()` (PR #662) works in isolation, **it is bypassed when `allowAttrs` is configured** — which is exactly the case with the `markuplint:a11y` preset.

### Root Cause

In `invalid-attr/index.ts`, the evaluation order is:

1. `allowAttrs` match → custom value check (line 186)
2. `disallowAttrs` match → disallow check (line 209)
3. **Only if neither matches** → `isValidAttr()` → `noUse` check (line 262)

The `markuplint:a11y` preset configures:

```json
"allowAttrs": [
  { "name": "tabindex", "value": { "enum": ["-1", "0"] } }
]
```

So for `<dialog tabindex="0">`:
- `tabindex` matches `allowAttrs` → step 1 runs
- `"0"` is in `["-1", "0"]` → passes validation
- **Step 3 (`noUse` check) is never reached**

### Verification

```js
// ✅ PASS: rule-only config detects the violation
config: { rules: { "invalid-attr": true } }
// → "The "tabindex" attribute is disallowed"

// ❌ FAIL: a11y preset equivalent misses the violation
config: { rules: { "invalid-attr": { options: { allowAttrs: [{ name: "tabindex", value: { enum: ["-1", "0"] } } ] } } } }
// → No violations (0)
```

### Fix Direction

The spec-level `noUse` check should take priority over `allowAttrs` configuration, since `noUse` represents an HTML spec violation that cannot be overridden by user configuration.
